accounts-db/accounts-hash-cache-tool/src/main.rs

```diff
@@ -225,22 +234,20 @@ fn cmd_diff_state(
     let path1 = value_t_or_exit!(subcommand_matches, "path1", String);
     let path2 = value_t_or_exit!(subcommand_matches, "path2", String);
     let num_bins = value_t_or_exit!(subcommand_matches, "bins", usize);
-    let bin_of_interest =
-        if let Some(bin_of_interest) = subcommand_matches.value_of("bin_of_interest") {
-            let bin_of_interest = bin_of_interest
-                .parse()
-                .map_err(|err| format!("argument 'bin-of-interest' is not a valid value: {err}"))?;
-            if bin_of_interest >= num_bins {
-                return Err(format!(
-                    "argument 'bin-of-interest' must be less than 'bins', \
-                     bins: {num_bins}, bin-of-interest: {bin_of_interest}",
-                ));
+
+    let bin_range_of_interest =
+        if let Some(bins) = values_of::<usize>(subcommand_matches, "bins_of_interest") {
+            match bins.len() {
+                1 => bins[0]..bins[0].saturating_add(1),
+                2 => bins[0]..bins[1],
+                _ => {
+                    unreachable!("invalid argument are given to bins_of_interest.")
+                }
             }
-            Some(bin_of_interest)
         } else {
-            None
+            0..usize::MAX
```

Comment:
.. is actually a different type - RangeFull, not Range. 
To make it work, we would need to update all the functions to take a range trait.
Since usize::MAX will not be used anyway, I think it should fine to just use Range here. 
WDYT?